common/components/collision-body.hpp

```diff
@@ -5,32 +5,60 @@
 #include <btBulletCollisionCommon.h>
 #include <btBulletDynamicsCommon.h>
 
+#include "multiton.hpp"
 #include "tec-types.hpp"
+#include "transforms.hpp"
 
 namespace tec {
+using PositionMap = Multiton<eid, Position*>;
+using OrientationMap = Multiton<eid, Orientation*>;
 ATTRIBUTE_ALIGNED16(struct) CollisionBody {
 
 	BT_DECLARE_ALIGNED_ALLOCATOR();
 
 	struct MotionState : public btMotionState {
-		MotionState() { this->transform.setIdentity(); }
+		MotionState(eid& _entity_id) : entity_id(_entity_id) { this->transform.setIdentity(); }
 		MotionState(MotionState&& other) noexcept :
-				transform(std::move(other.transform)), transform_updated(other.transform_updated) {}
+				transform(std::move(other.transform)), transform_updated(other.transform_updated),
+				entity_id(other.entity_id) {}
 
 		MotionState& operator=(MotionState&& other) noexcept {
 			transform_updated = other.transform_updated;
+			entity_id = other.entity_id;
 			return *this;
 		}
 
 		void getWorldTransform(btTransform& worldTrans) const { worldTrans = this->transform; }
 
-		void setWorldTransform(const btTransform& worldTrans) {
+		void setWorldTransform(const btTransform& worldTrans) override {
 			this->transform_updated = true;
 			this->transform = worldTrans;
+			if (entity_id != 0) {
+				if (PositionMap::Has(entity_id)) {
+					auto position = PositionMap::Get(entity_id);
+					btVector3 origin = transform.getOrigin();
+					position->value.x = origin.x();
+					position->value.y = origin.y();
+					position->value.z = origin.z();
+				}
+				if (OrientationMap::Has(entity_id)) {
+					auto orientation = OrientationMap::Get(entity_id);
+					btQuaternion rotation = transform.getRotation();
+					orientation->value.x = rotation.x();
+					orientation->value.y = rotation.y();
+					orientation->value.z = rotation.z();
+					orientation->value.w = rotation.w();
+					rotation.getEulerZYX(orientation->rotation.z, orientation->rotation.y, orientation->rotation.x);
+				}
+			}
```

Comment:
At the end of the physics system's `Update` method is a loop that maps entity IDs to transforms, if you moved all this there, then the motion state wouldn't have to track an entity ID, and all the positions and orientations would still get updated at the end of a step.

Reply:
I did have it there, but it felt odd.

The btMotionState docs recommended syncing the state in the method. The transform_updated bool was a hack to do this sync to the GameState.